### ISSUE TITLE
Having a `translates` method on an AR model blocks db creation and migration

### ISSUE THREAD

**Issue by mauriciopasquier** [shioyama/mobility#300]
I have a translated column on an ActiveRecord model like this:

```ruby
class ProfileType < ApplicationRecord
  extend Mobility
  translates :value

  # other stuff
end
```

If I try to create the db from zero

```bash
RAILS_ENV=test rails db:drop
RAILS_ENV=test rails db:create
rails aborted!
PG::ConnectionBad: FATAL:  database “myapp_test” does not exist
/some/path/app/models/profile_type.rb:3:in `<class:ProfileType>'
```

the line with `translates` blocks it. If I comment said line, everything works.

Using `rails db:reset`, which internally does `db:drop` followed by `db:create`, works. I think it is caching the column information, which of course it can't do when using separate commands.

I also hit this issue when trying to deploy for the first time with an empty database.

This is my configuration:

```ruby
Mobility.configure do |config|
  config.default_backend = :container
  config.accessor_method = :translates
  config.query_method = :i18n
  config.default_options[:fallbacks] = true
  config.default_options[:locale_accessors] = true
end
```

**Comment by shioyama** [comment]
I'm really not sure what Mobility would have to do with that.

You can see what it does in the model here:

https://github.com/shioyama/mobility/blob/2a8bd2d8644a360ed90bd6095d9ed70fe885fc77/lib/mobility/backends/active_record/container.rb#L81-L100

It's just calling `store` on the model with `:translations`. Can you create a test app that reproduces this?

**Comment by shioyama** [comment]
I think it's unlikely to be related to Mobility, and nobody else has noticed this, so I'll close for now. Feel free to re-open when you can demonstrate the issue.

**Comment by mauriciopasquier** [comment]
I tested a little more and got to the conclusion that:

1. It happens with `:container` backend (not, for example, with `key_value`).
2. It happens if a translated model has a "resource" page with [ActiveAdmin](https://github.com/activeadmin/activeadmin), which tries to load the corresponding model class maybe a little bit too early.

As this issue only appears with mobility-translated models, I assume its something we can fix here, but I might be wrong.

I made a demo app with only the required setup to fail [here](https://github.com/mauriciopasquier/mobility_active_admin_test/commits/master). To reproduce, try:

```bash
rails db:drop
rails db:create
rails db:reset
```
with, and without, `app/admin/things.rb`. These commands only fail after the last commit (the one wich changes to `:container`).

**Comment by shioyama** [comment]
Ok, the problem is [here](https://github.com/shioyama/mobility/blob/ae627c4cac341fedf9925c2d07277d4ffe11ea6c/lib/mobility/backends/active_record/container.rb#L53).

**Comment by shioyama** [comment]
So: the container backend dynamically checks the type of the column to see if it's JSON or JSONb, the based on that does some stuff. That check causes the issue here.

Sorry about that, this is indeed a Mobility issue, or at least Mobility *is* triggering it. I'll have to think about how to solve it.

**Comment by mauriciopasquier** [comment]
No problem! Glad you could pinpoint the issue.

**Comment by shioyama** [comment]
Fix in #300. If specs pass I'll merge and push a release.

**Comment by mauriciopasquier** [comment]
Awesome, thanks! I tested in my projects and it fixes the issue :blush:

**Comment by shioyama** [comment]
Released in 0.8.5. Thanks for the issue report!
